### PR TITLE
Make `@node` mandatory to generate Queries and Mutations.

### PR DESCRIPTION
**Pull request by MacondoExpress**
Change the way how `@node` behaves, `@node` is now required, and GraphQL Object types without the directive `@node` will no longer considered as a Neo4j Nodes representation. Queries and Mutations will be generated only for types with the `@node` directive.

### CODE REVIEW

**Review by angrykoala — APPROVED** [review]

File: packages/graphql/src/schema-model/parser/parse-attribute.ts [hunk 1]
```diff
@@ -115,7 +115,8 @@ function parseTypeNode(
                 return new InterfaceType(typeNode.name.value, isRequired);
             } else if (isInput(definitionCollection, typeNode.name.value)) {
                 return new InputType(typeNode.name.value, isRequired);
-            } else {
+            }
+            else {
```
Comment (mjfwebb) [inline packages/graphql/src/schema-model/parser/parse-attribute.ts]:
```suggestion
            } else {
```

File: packages/apollo-federation-subgraph-compatibility/src/type-defs.ts [hunk 2]
```diff
@@ -42,7 +42,7 @@ export const typeDefs = gql`
 
     directive @custom on OBJECT
 
-    type Product @custom @key(fields: "id") @key(fields: "sku package") @key(fields: "sku variation { id }") {
+    type Product @node @custom @key(fields: "id") @key(fields: "sku package") @key(fields: "sku variation { id }") {
```
Comment (MacondoExpress) [inline packages/apollo-federation-subgraph-compatibility/src/type-defs.ts:45]: @darrellwarde can you verify the changes made here?

Comment (darrellwarde) [reply packages/apollo-federation-subgraph-compatibility/src/type-defs.ts:45]: No? 😆 It looks fine to me! It's a node in our database right?!

File: packages/graphql/tests/schema/issues/1575.test.ts [hunk 3]
```diff
@@ -78,15 +78,6 @@ describe("https://github.com/neo4j/graphql/issues/1575", () => {
               node: Foo!
             }
 
-            input FooOptions {
```
Comment (darrellwarde) [inline packages/graphql/tests/schema/issues/1575.test.ts]: I don't know how this ended up in here?

**Review by darrellwarde — APPROVED** [review]
LGTM!!

File: packages/apollo-federation-subgraph-compatibility/src/type-defs.ts (hunk 2 above)
Comment (MacondoExpress) [reply packages/apollo-federation-subgraph-compatibility/src/type-defs.ts:45]: It is!

**Pull request merged**